coriolis/tests/osmorphing/test_debian.py

```diff
@@ -0,0 +1,297 @@
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_write_file_sudo')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_read_file')
+    def test_disable_predictable_nic_names(
+            self, mock_read_file, mock_write_file_sudo, mock_exec_cmd_chroot,
+            mock_test_path, mock_grub2_cfg_editor):
+        mock_test_path.return_value = True
+
+        self.morpher.disable_predictable_nic_names()
+
+        mock_test_path.assert_called_once_with(
+            self.os_root_dir + '/etc/default/grub')
+        mock_grub2_cfg_editor.assert_called_once_with(
+            mock_read_file.return_value.decode.return_value)
+        mock_grub2_cfg_editor.return_value.append_to_option.assert_has_calls(
+            [
+                mock.call(
+                    "GRUB_CMDLINE_LINUX_DEFAULT",
+                    {"opt_type": "key_val", "opt_key": "net.ifnames",
+                     "opt_val": 0}),
+                mock.call(
+                    "GRUB_CMDLINE_LINUX_DEFAULT",
+                    {"opt_type": "key_val", "opt_key": "biosdevname",
+                     "opt_val": 0}),
+                mock.call(
+                    "GRUB_CMDLINE_LINUX",
+                    {"opt_type": "key_val", "opt_key": "net.ifnames",
+                     "opt_val": 0}),
+                mock.call(
+                    "GRUB_CMDLINE_LINUX",
+                    {"opt_type": "key_val", "opt_key": "biosdevname",
+                     "opt_val": 0})
+            ]
+        )
+        mock_read_file.assert_called_once_with(
+            self.os_root_dir + '/etc/default/grub')
+        mock_write_file_sudo.assert_called_once_with(
+            "etc/default/grub", mock_grub2_cfg_editor.return_value.dump())
+        mock_exec_cmd_chroot.assert_called_once_with("/usr/sbin/update-grub")
+
+    @mock.patch('coriolis.utils.Grub2ConfigEditor')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_write_file_sudo')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_read_file')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_test_path')
+    def test_disable_predictable_nic_names_no_test_path(
+            self, mock_test_path, mock_read_file,
+            mock_write_file_sudo, mock_exec_cmd_chroot, mock_grub2_cfg_editor):
+
+        mock_test_path.return_value = False
+
+        self.morpher.disable_predictable_nic_names()
+
+        mock_read_file.assert_not_called()
+        mock_write_file_sudo.assert_not_called()
+        mock_exec_cmd_chroot.assert_not_called()
+        mock_grub2_cfg_editor.assert_not_called()
+
+    def test_get_update_grub2_command(self):
+        result = self.morpher.get_update_grub2_command()
+        self.assertEqual(result, "update-grub")
+
+    @mock.patch('coriolis.osmorphing.debian.StringIO')
```

Comment:
The test case can be written without `StringIO` being mocked.

For example, without mocking anything, you can do assertion:
```python
expected_result = (
    "\n"
    "auto lo\n"
    "iface lo inet loopback\n"
    "\n\n\n"
    "auto eth0\n"
    "iface eth0 inet dhcp\n"
    "\n\n\n"
    "auto eth1\n"
    "iface eth1 inet dhcp\n"
    "\n\n"
)
self.assertEqual(result, expected_result)
```